cmake/recipes/boost.cmake

```diff
@@ -33,6 +33,9 @@ CPMAddPackage(
 set(BOOST_SOURCE ${boost_SOURCE_DIR})
 set(Boost_POPULATED ON)
 
+# Only build the following Boost libs
+set(BOOST_LIBS_OPTIONAL "" CACHE STRING "Boost libs to be compiled" FORCE)
```

Comment:
Rather than updating `boost-cmake` to Boost 1.83.0, I removed all libs because AMGCL only requires `property_tree` which is header only.